api/types/hardware.go

```diff
@@ -12,8 +12,50 @@ type GPU struct {
 }
 
 type HardwareSpec struct {
-	GPU     GPU                  `json:"gpu"`
-	CPU     HardwareRequestRange `json:"cpu"`
-	RAM     HardwareRequestRange `json:"ram"`
-	Storage HardwareRequestRange `json:"storage"`
+	GPU GPU                  `json:"gpu"`
+	CPU HardwareRequestRange `json:"cpu"`
+	RAM HardwareRequestRange `json:"ram"`
+	HDD HardwareRequestRange `json:"hdd"`
+}
+
+func SetSpecDefaultValues(spec HardwareSpec) HardwareSpec {
```

Comment:
Maybe do one of either mutating the state or returning a new `HardwareSpec` type but not both. That seems like it would cause issues. Maybe make a copy and return it?

Reply:
Yeah we can do that!

Reply:
This got marked as outdated but still un resolved

Reply:
Feedback hasn’t been addressed yet

Reply:
Resolved